contracts/Account_base.cairo

```diff
@@ -0,0 +1,221 @@
+%lang starknet
+
+from starkware.cairo.common.registers import get_fp_and_pc
+from starkware.starknet.common.syscalls import get_contract_address
+from starkware.cairo.common.signature import verify_ecdsa_signature
+from starkware.cairo.common.cairo_builtins import HashBuiltin, SignatureBuiltin
+from starkware.starknet.common.syscalls import call_contract, get_caller_address, get_tx_signature
+from starkware.cairo.common.hash_state import (
+    hash_init, hash_finalize, hash_update, hash_update_single
+)
+
+#
+# Structs
+#
+
+struct AccountMessage:
+    member sender: felt
+    member to: felt
+    member selector: felt
+    member calldata: felt*
+    member calldata_size: felt
+    member nonce: felt
+end
+
+#
+# Storage
+#
+
+@storage_var
+func Account_current_nonce() -> (res: felt):
+end
+
+@storage_var
+func Account_public_key() -> (res: felt):
+end
+
+#
+# Guards
+#
+
+func Account_assert_only_self{
+        syscall_ptr : felt*, 
+        pedersen_ptr : HashBuiltin*,
+        range_check_ptr
+    }():
+    let (self) = get_contract_address()
```

Comment:
I'm wondering if this needs the `Account` prefix, since we're not exporting it and it's not necessarily Account-centric

Reply:
it _is_ exported, we're just not using it but someone else could

Reply:
agree it's not account-specific